onvm/onvm_mgr/onvm_init.c

```diff
@@ -452,50 +446,6 @@ init_shared_sem(void) {
         }
 }
 
-/**
- * Set up the DPDK rings which will be used to pass packets, via
- * pointers, between the multi-process server and NF processes.
- * Each NF needs one RX queue.
- */
-static int
-init_shm_rings(void) {
-        unsigned i;
-        unsigned socket_id;
-        const char *rq_name;
-        const char *tq_name;
-        const char *msg_q_name;
-        const unsigned ringsize = NF_QUEUE_RINGSIZE;
-        const unsigned msgringsize = NF_MSG_QUEUE_SIZE;
-
-        // use calloc since we allocate for all possible NFs
-        // ensure that all fields are init to 0 to avoid reading garbage
-        // TODO plopreiato, move to creation when a NF starts
```

Comment:
Looks like Phil already wanted this done, the stats don't look negatively affected which is good. My only concern is that it's nice to init the rings in the beginning, so that there is no overhead for each additional startup. Imagine a situation where we wanted a very high number of short NF tasks, initializing at the beginning of each startup might lower the efficiency of creating those NFs overall.

Reply:
Thanks for the comment. I agree, It does create a little bit of extra overhead on startup. However, I would argue that it's negligible in this environment though, since we still need a manager <-> nf handshake, in which we `sleep(1)` in order to get NF specific structures set up. The NF sends a message to the manager [here](https://github.com/sdnfv/openNetVM/blob/master/onvm/onvm_nflib/onvm_nflib.c#L414). The manager then allocates the ring structures along with the NF ID [upon receiving the message](https://github.com/sdnfv/openNetVM/blob/master/onvm/onvm_mgr/onvm_nf.c#L175) 
Reading over this though, it'd be nice to try and optimize out the handshake process.